### DISCUSSION TITLE
Klipperscreen installed successfully but does not run

### DISCUSSION THREAD

**Discussion by pdsavas** [KlipperScreen/KlipperScreen#1102] [Q&A]
I have a system running on a Pi Zero 2 with a Waveshare 3.5" screen connected to the GPIO pins. I've installed Klipperscreen successfully (using KIAUH), but when the system starts up, the Pi boots to a command prompt, not Klipperscreen. I've never had to do anything to get a Pi-based install to start with the GUI, so I'm a bit flummoxed by this. I've tried uninstalling and reinstalling Klipperscreen to no avail.

In another thread, I found this line of code to start Klipperscreen:

sh -c "$HOME/.KlipperScreen-env/bin/python $HOME/KlipperScreen/screen.py"

When I tried it from a command prompt, this is what I got back:

Unable to init server: Could not connect: Connection refused
Unable to init server: Could not connect: Connection refused
2023-08-29 00:57:53,032 [screen.py:main()] - Python version: 3.9
2023-08-29 00:57:53,033 [screen.py:main()] - KlipperScreen version: v0.3.5-10-gfba15bd
Unable to init server: Could not connect: Connection refused
2023-08-29 00:57:53,036 [screen.py:__init__()] - Gtk couldn't be initialized. Use Gtk.init_check() if you want to handle this case.
Traceback (most recent call last):
  File "/home/pi/KlipperScreen/screen.py", line 97, in __init__
    super().__init__(title="KlipperScreen")
  File "/home/pi/.KlipperScreen-env/lib/python3.9/site-packages/gi/overrides/Gtk.py", line 519, in __init__
    raise RuntimeError(
RuntimeError: Gtk couldn't be initialized. Use Gtk.init_check() if you want to handle this case.

Would anyone be able to tell me what's going on and how I can fix this?

Thanks!

**Comment by alfrix** [comment]
Connection refused sounds like:
https://klipperscreen.readthedocs.io/en/latest/Troubleshooting/VC_ERROR/

**Reply by pdsavas** [reply]
I followed the directions on that page and there was no change. :(

This is what came up on the screen, though:

```
[   24.025028] rc.local[803]: fbcp: error while loading shared libraries: libbcm_host.so: cannot open shared object file; No such file or directory
Debian GNU/Linux 11 kingroonpi ttyl
```

Does that help at all?

**Reply by alfrix** [reply]
fbcp is part of the screen driver stack, you may have missing packages, how did you install the screen?

**Reply by pdsavas** [reply]
The screen itself sits on the GPIO pins of the Pi, and then I installed Waveshare's drivers as per their Website. If you're referring to how installed Klipperscreen, I used KIAUH.

**Reply by alfrix** [reply]
do:
`dpkg-query --search libbcm_host.so`
in my machine it shows these two packages that contain said library:
```
libraspberrypi0:arm64: /usr/lib/aarch64-linux-gnu/libbcm_host.so.0
libraspberrypi-dev: /usr/lib/aarch64-linux-gnu/libbcm_host.so
```
libraspberrypi0 and libraspberrypi-dev
ensure those are insalled

the other thing that can be possible is that there is an issue in the distro, in my case since i'm using the 64-bit i would try the 32bit version

**Reply by pdsavas** [reply]
Both of the libraries were installed:

`pi@kingroonpi:~ $ dpkg-query --search libbcm_host.so
libraspberrypi0:arm64: /usr/lib/aarch64-linux-gnu/libbcm_host.so.0
libraspberrypi-dev: /usr/lib/aarch64-linux-gnu/libbcm_host.so
`
Are there any other diagnostic steps I can take to try and figure out what's going on?

I am assuming KIAUH installed the 64-bit version of Klipperscreen because I'm using a 64-bit OS; how would I go about installing the 32-bit version?

Thanks.

**Reply by alfrix** [reply]
there are no 64 - 32 bits of klipperscreen, i was talking trying the 32 bit OS

**Reply by pdsavas** [reply]
Well, I can certainly reflash the card with the 32-bit version of the Mainsail image in the Raspberry Pi Imager and let you know if that corrects the problem:
![image](https://github.com/KlipperScreen/KlipperScreen/assets/46990222/7c14553a-6e69-402e-aafc-3d9ac1c7f6df)

**Reply by pdsavas** [reply]
Well, no change with 32-bit OS; Klipperscreen is still not appearing on the Pi screen, and now, instead of seeing a text message with the login prompt, I just see a flashing cursor where the login prompt was when I had the 64-bit version of the OS installed.

What now?

**Reply by alfrix** [reply]
i cannot help without seeing the new logs

**Reply by pdsavas** [reply]
Which log files do you need to see?
